### DISCUSSION TITLE
Allow trading rewards and adding funds to an LM pool in one action

### DISCUSSION THREAD

**Discussion by John-Gee** [DeFiCh/app#1010] [Ideas]
Hello,

similarly to the proposition by @nippur72 to allow a double swap in one action, I think it would be nice to be able to use rewards to fund an LM pool in one action. For now we have to wait for the first transaction on the DEX to be complete before adding funds, but also we pay 2 transaction fees (one for each) when the whole thing could have been relayed in a single message. Not that it's a problem today, but it would also reduce workload on the network. It also would reduce potential slippage problems between the 2 transactions: conversion of the funds and the funding, which can leave one side smaller than needed to fund 100%.

Thank you!

**Comment by thedoublejay** [comment]
Will be tracked on DeFi Wallet repo - https://github.com/DeFiCh/wallet/issues/1204
